docs/dev/relay_pass_manager.rst

```diff
@@ -0,0 +1,606 @@
+    };
+
+``pass_info`` is identical to what we just described in the module pass.
+``pass_func`` takes a function for optimization, it also needs a module as we
+may use it for reporting errors. A function could be annotated with
+"SkipOptimization" so that it will be ignored during optimization.
+
+Sequential Passes
+^^^^^^^^^^^^^^^^^
+
+SequentialPass is similar to Pytorch nn.Sequential that contains a host of
+passes for execution. It also has a disabled list that contains the list of
+disabled passes. For example, we may have a configuration which performs all
+passes at ``opt_level=3`` with some disabled passes using disabled_pass=xx
+provided by ``PassContext``. Now using ``SequentailPass``, we could glob all
+passes at ``opt_level=3`` and exclude those in disabled.
+
+.. code:: c++
+
+    class SequentialPassNode : PassNode {
+      PassInfo pass_info;
+      // Passes need to be executed.
+      Array<Pass> passes;
+      // Passes are disabled during optimization
+      std::vector<std::string> disabled;
+      bool PassEnabled(const PassInfo& info) const;
+      Module operator()(const Module& mod, const PassContext& pass_ctx) const final;
+    };
+
+Only a few passes currently in Relay is put in this group. For example,
+``FoldScaleAxis`` requires to dispatch ``ForwardFoldScaleAxis`` and
+``BackwardFoldScaleAxis`` internally. In addition, ``BackwardFoldScaleAxis`` is
+recommended to be fulfilled first. This pass, hence, is an ideal candidate for
+``SequentailPass``.
+
+The following code shows how individual passes in a sequential pass is invoked.
+Essentially, we sequentially execute each pass in a sequential pass using the
+order that they were appended to the pass list without the consideration of
+their orders. The pass phase ordering problem might be considered in the future.
+
+.. code:: c++
+
+    Module SequentialNode::operator()(const Module& module,
+                                      const PassContext& pass_ctx) const {
+      Module mod = module;
+      for (const Pass& pass : passes) {
+        CHECK(pass.defined()) << "Found undefined pass for optimization.";
+        const PassInfo& pass_info = pass->Info();
+        if (!PassEnabled(pass_info))  continue;
+        for (const auto& it : pass_info->required) {
+          const auto* name = it.as<tvm::ir::StringImm>();
+          CHECK(name);
+          mod = GetPass(name->value)(mod, pass_ctx);
+        }
+        mod = pass(mod, pass_ctx);
+      }
+      return mod;
+    }
+
+Upon the invocation of a pass, we first check if this pass is enabled. This is
+done by first checking if the pass is explicitly disabled by a user, followed by
+inspecting if it is specified as a required pass by the user. If it is still
+indetermined that whether this pass is enabled, its ``opt_level`` will be
+checked. This pass will be enabled and therefore executed only when its
+optimization level not less than the configured optimization level in the pass
+context.
```

Comment:
So, opt_level will be ignored, if user just specifies that I want to run Pass A, correct? Just like gcc -opt_A works even if we dont provide -O3 etc.

Reply:
yes